src/config_store.rs

```diff
@@ -0,0 +1,197 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+
+
+#[derive(Debug)]
+pub struct ConfigStore {
+    file_name: String,
+    config: ConfigNameSpaces
```

Comment:
I have experimented with that before, but it messed with de/serialization and made return types awkward. Will try again.